### ISSUE TITLE
User still shown as sharing with me after I've ignored them

### ISSUE THREAD

**Issue by goobertron** [diaspora/diaspora#7755]
My understanding has always been that by ignoring someone, I break the sharing relationship, and that this will thereby stop my posts (including public posts) from appearing in their stream, although they could still view my public posts by going to SPV for a post.

Indeed, when I last updated the tutorials on the website, @SuperTux88 advised me to write:

> [Ignoring a user] will prevent any of their posts from appearing in your stream, *and they will no longer be able to like or comment on your posts*. They will, however, still be able to reshare your posts, comment on reshares of your posts, and their comments on posts by other people which appear in your stream will still be visible to you.

However, I just tested this, ignoring Account B on a different pod to Account A, and was able to comment from Account B on a post I made from Account A after ignoring. So it seems that nothing changes any more in the sharing relationship when ignoring someone. (This used to be the case, as it produced an anomaly when starting to share once again with an account that you had previously been ignoring: #4757.)

**Comment by SuperTux88** [comment]
If I understand your question correctly that's normal. When A and B are on different pods, B doesn't know about A ignoring him (this will change in the future, parts for this are already done in the federation). Therefore it looks like B can still comment, but as soon as the comment reaches the pod of A a delete message is sent back and the comment from B is deleted again.

About the sharing state: In the past, your pod completely deleted the sharing state (so when B shared with A, A didn't know about this anymore, but B was still sharing with A), which created weird broken inconsistency states when you stopped ignoring somebody again. That was fixed a while ago, so A still remembers the correct sharing state of B sharing with A (and also displays it to you).

So if I understand your "problem" correctly this works everything as intended and it's not a bug.

**Comment by goobertron** [comment]
Thanks for your reply.

It sounds as though things are working as they should, then. I tested it just before opening this issue, and was sure that I saw the comment from B on account A. However I've just repeated the test a couple of times since reading your response and the comment is actually deleted, just as you describe.

It sounds as though my understanding was at fault. I thought the bug in the earlier version was that it prevented the sharing relationship from B to A from being re-established when A removed the block. But it seems that the sharing from B to A wasn't (or wasn't intended to be) broken in the first place.

I'll therefore close this ticket. Thanks again for the explanation.
